packages/circuit-ui/components/DateInput/DateInput.tsx

```diff
@@ -435,20 +372,80 @@ export const DateInput = forwardRef<HTMLInputElement, DateInputProps>(
                 }
               })}
             </div>
-            <IconButton
-              ref={calendarButtonRef}
-              type="button"
-              icon={CalendarIcon}
-              variant="secondary"
-              onClick={openCalendar}
-              className={classes['calendar-button']}
-              disabled={disabled || readOnly}
-              aria-expanded={open}
-              aria-haspopup="true"
-              aria-controls={dialogId}
+            <Popover
+              ref={dialogRef}
+              id={dialogId}
+              isOpen={open}
+              onToggle={closeCalendar}
+              aria-labelledby={headlineId}
+              className={clsx(!isMobile && classes.popover)}
+              offset={4}
+              placement={placement}
+              closeButtonLabel={closeCalendarButtonLabel}
+              locale={locale}
+              component={() => (
+                <IconButton
+                  ref={calendarButtonRef}
+                  type="button"
+                  icon={CalendarIcon}
+                  variant="secondary"
+                  onClick={toggleCalendar}
+                  className={classes['calendar-button']}
+                  disabled={disabled || readOnly}
+                  aria-expanded={open}
+                  aria-haspopup="true"
+                  aria-controls={dialogId}
+                >
+                  {calendarButtonLabel}
+                </IconButton>
+              )}
             >
-              {calendarButtonLabel}
-            </IconButton>
+              {() => (
+                <div className={clsx(!isMobile && classes.content)}>
+                  <header className={classes.header}>
+                    <Headline as="h2" size="m" id={headlineId}>
+                      {label}
+                    </Headline>
+                  </header>
+
+                  <Calendar
+                    className={classes.calendar}
+                    onSelect={handleSelect}
+                    selection={selection}
+                    minDate={minDate}
+                    maxDate={maxDate}
+                    locale={locale}
+                    firstDayOfWeek={firstDayOfWeek}
+                    modifiers={modifiers}
+                    prevMonthButtonLabel={prevMonthButtonLabel}
+                    nextMonthButtonLabel={nextMonthButtonLabel}
+                  />
+                  <Hr className={classes.divider} />
```

Comment:
What prompted you to add the Hr component instead of the previous border-top style?

I see two issues with the current implementation:

- The divider shouldn't be visible when the buttons aren't rendered.
- The Hr element has semantic meaning that isn't necessary here. I'd remove it with `role="presentation"`.